service/src/main/java/org/databiosphere/workspacedataservice/recordsink/RawlsRecordSink.java

```
@@ -86,23 +85,32 @@ public void deleteBatch(RecordType recordType, List<Record> records) throws Batc
   }
 
   @Override
-  public void finalizeBatchWrite(BatchWriteResult result) {
-    // currently a no-op
-    // TODO(AJ-1669): do pubsub here, maybe do GCS cleanup here (assuming a file was reused
-    //   throughout)
+  public void close() {
+    if (jsonWriter != null) {
+      try {
+        jsonWriter.close();
+        jsonWriter = null; // reset the writer; they aren't reusable
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    publishToPubSub(
+        importDetails.collectionId(),
+        requireNonNull(importDetails.userEmail()),
```

I'm still getting used to the nullable stuff, why is this explicitly called out as require non null? aren't they all required to be non null unless marked with `@nullable`?

---

`importDetails.userEmail()` is marked `@Nullable` at its [declaration](https://github.com/DataBiosphere/terra-workspace-data-service/blob/main/service/src/main/java/org/databiosphere/workspacedataservice/dataimport/ImportDetails.java#L9)

While `publishToPubSub` expects to receive a `@NonNull String` for this argument (as is now the default expectation).

`Objects.requireNonNull` has the nice side effects of 

1. throwing an exception if given a `null` object
2. returning a `@NonNull` version of what it is given, thereby satisfying the `@NonNull` requirement of `publishToPubSub`

Since `publishToPubSub` won't work correctly if given `null`, I'm ok with this crashing right here instead of letting `null` through, so `requireNonNull` does the trick here.